### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/varnish-cache/actions/runs/10295267562#summary-28494690560

The action will report that the changes in this PR would make it happy: https://github.com/jsoref/varnish-cache/actions/runs/10301871471#summary-28514346569

### CODE REVIEW

File: doc/changes.rst [hunk 1]
```diff
@@ -6468,7 +6468,7 @@ Build system
    wider range of platforms.
 
 -  Compatibility shims for clock\_gettime() are now correctly applied
-   where needed, allowing Varnish to build on MacOS X.
+   where needed, allowing Varnish to build on Mac OS X.
```
Comment (jsoref) [inline doc/changes.rst:6471]: I should have peeled this one to a commit named `mac os` -- note that this item predates the brand transitions between `Mac OS X` and `macOS`

**Comment by bsdphk** [comment]
Is it possible you could structure this as one PR per directory, that would make it a lot more manageable for us ?

**Comment by jsoref** [comment]
@bsdphk: done

**Comment by nigoroll** [comment]
@bsdphk should we work on this incrementally by cherry-picking what is clearly uncontroversial?

**Comment by jsoref** [comment]
Oh, sorry. I misread. You wanted PRs not commits. I can do that, but I've probably run out of time until Wednesday night.

**Comment by jsoref** [comment]
@bsdphk / @nigoroll I've split out the other items into distinct PRs:

* #4156 
* #4157
* #4158
* #4159
* #4160
* #4161
* #4162

I'm also on irc if that's helpful.

**Comment by asadsa92** [comment]
Seems like all splits has been merged.
Should we add the spell-check job to the pipeline? or just close the issue?

**Comment by jsoref** [comment]
I tightened up some patterns and looked more closely at the remaining list of tokens that weren't in my base dictionary and found a couple of other typos.

I can split out the `bin` and `doc` changes into distinct PRs again.

If you would like to pull in check-spelling, you'll want to review the [word list](https://github.com/jsoref/varnish-cache/blob/spell-check/.github/actions/spelling/expect.txt) to see if there are any other items that are also typos (I had missed at least `implicit` and `catastrophe`).

**Comment by nigoroll** [comment]
Sorry, the merge has closed the issue, but we still need to discuss if we want to merge the spell-checking, so I opened a new issue

**Pull request merged**